### PR TITLE
load table of contents from body and scroll

### PR DESCRIPTION
**Pull request by ilya-zhuravel**

### CODE REVIEW

**Review by norisalsaadie — APPROVED** [review]
Not too familiar with the code but I don't spot any obvious concerns

File: src/app/containers/manuscript/manuscript-toc.tsx [hunk 1]
```diff
@@ -14,10 +17,26 @@ export interface ManuscriptTOCProps {
...
 
   const articleInfo = useSelector(getArticleInformation);
   const journalMeta = useSelector(getJournalMeta);
+  const manuscriptBodyTOC = useSelector(getManuscriptBodyTOC);
+  const toc: TableOfContent = [
+    { level: 1, title: 'Title', id: 'title' },
+    { level: 1, title: 'Abstract', id: 'abstract' },
+    { level: 1, title: 'Impact statement', id: 'impactStatement' },
+    ...manuscriptBodyTOC,
+    { level: 1, title: 'Acknowledgements', id: 'acknowledgement' }
```
Comment (NuclearRedeye) [inline src/app/containers/manuscript/manuscript-toc.tsx:31]: id should be 'acknowledgements'

File: src/app/containers/manuscript/manuscript-toc.tsx [hunk 2]
```diff
@@ -14,10 +17,26 @@ export interface ManuscriptTOCProps {
...
   const { tocOpen, handleTocToggle } = props;
 
   const articleInfo = useSelector(getArticleInformation);
   const journalMeta = useSelector(getJournalMeta);
+  const manuscriptBodyTOC = useSelector(getManuscriptBodyTOC);
+  const toc: TableOfContent = [
+    { level: 1, title: 'Title', id: 'title' },
+    { level: 1, title: 'Abstract', id: 'abstract' },
+    { level: 1, title: 'Impact statement', id: 'impactStatement' },
+    ...manuscriptBodyTOC,
```
Comment (NuclearRedeye) [inline src/app/containers/manuscript/manuscript-toc.tsx:30]: There should be an entry here for 'main body' I think, which takes you to the start of the main body.

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]
Some changes needed, also can break it by...
1. Select 'Acknowledgements'
2. Try and click anything else.
I think that might be because the ID is incorrect, but feel that it shouldn't stop the whole thing from working!

I also can't use the paragraph menu any more to create new headings, etc.

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]
Good work, that's fixed those previous issues. I've found another issue which I think might be worth having a quick look at here, but if it turns out to be a rabbit hole we can address that under a new issue. Basically, if you copy/paste some content the duplicate entries in the TOC always take you to the first section, so I assume when you copy/paste it is also copying the ID and hence that causes the navigation to trip up a little.

**Review by NuclearRedeye — APPROVED** [review]
Tested locally and this is working well now.

**Pull request merged**